// @langchain/anthropic/dist/tools/index.cjs
const require_memory = require('./memory.cjs');
const require_webSearch = require('./webSearch.cjs');
const require_webFetch = require('./webFetch.cjs');
const require_toolSearch = require('./toolSearch.cjs');
const require_textEditor = require('./textEditor.cjs');
const require_computer = require('./computer.cjs');
const require_codeExecution = require('./codeExecution.cjs');
const require_bash = require('./bash.cjs');
const require_mcpToolset = require('./mcpToolset.cjs');

//#region src/tools/index.ts
const tools = {
	memory_20250818: require_memory.memory_20250818,
	webSearch_20250305: require_webSearch.webSearch_20250305,
	webFetch_20250910: require_webFetch.webFetch_20250910,
	toolSearchRegex_20251119: require_toolSearch.toolSearchRegex_20251119,
	toolSearchBM25_20251119: require_toolSearch.toolSearchBM25_20251119,
	textEditor_20250728: require_textEditor.textEditor_20250728,
	computer_20251124: require_computer.computer_20251124,
	computer_20250124: require_computer.computer_20250124,
	codeExecution_20250825: require_codeExecution.codeExecution_20250825,
	bash_20250124: require_bash.bash_20250124,
	mcpToolset_20251120: require_mcpToolset.mcpToolset_20251120
};

//#endregion
exports.tools = tools;
